### ISSUE TITLE
[Metrics] Use tempfile module instead of '/tmp'

### ISSUE THREAD

**Issue by simon-mo** [ucbrise/clipper#448]
`/tmp` is not Windows compatible. Use `tempfile` module instead:
```
./docker/docker_metric_utils.py:def ensure_clipper_tmp():
./docker/docker_metric_utils.py:    Make sure /tmp/clipper directory exist. If not, make one.
./docker/docker_metric_utils.py:        os.makedirs('/tmp/clipper')
./docker/docker_metric_utils.py:    ensure_clipper_tmp()
./docker/docker_metric_utils.py:    with open('/tmp/clipper/prometheus.yml', 'w') as f:
./docker/docker_metric_utils.py:            '/tmp/clipper/prometheus.yml': {
./docker/docker_metric_utils.py:    with open('/tmp/clipper/prometheus.yml', 'r') as f:
./docker/docker_metric_utils.py:    with open('/tmp/clipper/prometheus.yml', 'w') as f:
./docker/docker_metric_utils.py:    with open('/tmp/clipper/prometheus.yml', 'r') as f:
./docker/docker_metric_utils.py:    with open('/tmp/clipper/prometheus.yml', 'w') as f:
./clipper_admin.py:CLIPPER_TEMP_DIR = "/tmp/clipper"
./metric/server.py:        'redis-server', '--unixsocket', '/tmp/redis.sock', '--daemonize', 'yes'
./metric/config.py:UNIX_SOCKET_PATH = '/tmp/redis.sock'
```

**Comment by simon-mo** [comment]
I'll fix occurrences in `./docker/docker_metric_utils.py`, the rest is not user-facing.

**Comment by simon-mo** [comment]
Closed by #503
